spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/store/channel/PostgresChannelMessageStoreQueryProvider.java

```diff
@@ -27,30 +27,69 @@ public class PostgresChannelMessageStoreQueryProvider implements ChannelMessageS
 
 	@Override
 	public String getPollFromGroupExcludeIdsQuery() {
-		return SELECT_COMMON
-				+ "and %PREFIX%CHANNEL_MESSAGE.MESSAGE_ID not in (:message_ids) "
-				+ "order by CREATED_DATE, MESSAGE_SEQUENCE LIMIT 1 FOR UPDATE SKIP LOCKED";
+		return """
+				delete
+				from %PREFIX%CHANNEL_MESSAGE
+				where CTID = (select CTID
+								from %PREFIX%CHANNEL_MESSAGE
+								where %PREFIX%CHANNEL_MESSAGE.GROUP_KEY = :group_key
+								and %PREFIX%CHANNEL_MESSAGE.REGION = :region
+								and %PREFIX%CHANNEL_MESSAGE.MESSAGE_ID not in (:message_ids)
+							order by CREATED_DATE, MESSAGE_SEQUENCE
+							limit 1)
```

Comment:
I understand that we don't need `FOR UPDATE`. But how about `SKIP LOCKED`?
WDYT?

Reply:
Hmm. I don't know. I think I'll run my performance tests if it makes a difference. Keep you posted.

Reply:
Thanks. 
Would you mind coming back with results as soon as possible?
We have a release today, so would be great to have this merged.